File: misc/theme.js
function hexToRGB(hex) {
  const [r, g, b] = [hex.slice(1, 3), hex.slice(3, 5), hex.slice(5, 7)]
    .map((hex) => parseInt(hex, 16) / 255)
    .map((num) => num.toFixed(8));

  return `Color::rgb(${r}, ${g}, ${b})`;
}

// paste a theme here from either:
// - https://github.com/saadeghi/daisyui/blob/master/src/theming/themes.js
// - https://daisyui.com/theme-generator/
// and then run `node ./misc/theme.js`
const themes = {
  dracula: {
    primary: "#ff79c6",
    secondary: "#bd93f9",
    accent: "#ffb86c",
    neutral: "#414558",
    "base-100": "#282a36",
    info: "#8be9fd",
    success: "#50fa7b",
    warning: "#f1fa8c",
    error: "#ff5555",
  },
};

for (const themeName of Object.keys(themes)) {
  const mappedColors = Object.entries(themes[themeName])
    .map(([key, value]) => [key.replace("-", ""), hexToRGB(value)])
    .map(([key, value]) => `\t${key}: ${value},\n`)
    .join("");

  const output = `const ${themeName.toUpperCase()}: Theme = Theme {\n${mappedColors}};`;
  console.log(output);
}

File: src/theme.rs
use bevy::prelude::*;

pub struct Theme {
    pub Primary: Color,
    pub Secondary: Color,
    pub Accent: Color,
    pub Neutral: Color,
    pub Info: Color,
    pub Success: Color,
    pub Warning: Color,
    pub Error: Color,
}

// https://github.com/dracula/dracula-theme
// https://github.com/saadeghi/daisyui/blob/master/src/theming/themes.js#L117
pub const MAIN_THEME: Theme = Theme {
    Primary: Color::rgb(1., 0.4745, 0.7764),
    Secondary: Color::rgb(0.7411, 0.5764, 0.9764),
    Accent: Color::rgb(1., 0.7215, 0.4235),
    Neutral: Color::rgb(0.2549, 0.2705, 0.3450),
    Info: Color::rgb(0.5450, 0.9137, 0.9921),
    Success: Color::rgb(0.3137, 0.9803, 0.4823),
    Warning: Color::rgb(0.9450, 0.9803, 0.5490),
    Error: Color::rgb(1., 0.3333, 0.3333),
};
